cpp/include/ray/api/ray_runtime_holder.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <ray/api/ray_runtime.h>
+
+namespace ray {
+namespace api {
+
+struct RayRuntimeHolder {
```

Comment:
Is this a user-accessible struct? Why does it have to be in `ray::api`? Could you please document the functionality of this struct?

Reply:
The ray runtime object used by some classes, such as Ray class, ActorHandle class, ObjectRef class, so move it into  RayRuntimeHolder, make it independent and unique, then other classed can use it conveniently, no circular references anymore.

Reply:
Since it's not part of Ray API, could you move it out of `ray::api`? We don't have the constraint that code in `ray::api` couldn't reference code in `ray::runtime` or other namespaces. We have this constraint in Java because there are two modules, but I don't think we need to compile `ray::api` and `ray::runtime` into separate .so files.

Reply:
Right, it's need to move out from ray::api.

Reply:
it's still in `ray::api`. Better to put it in `ray::runtime` or `ray::internal`. I'm not sure what's the preferred namespace for non API code.

Reply:
Have move it into ray::internal.